### PR TITLE
Music Survey - Jing & Cornelia 

### PR DESCRIPTION
**Pull request by lunek1**
## Netlify link
https://jing-cornelia-music-survey.netlify.app/

## Collaborators
[jingh999]

### CODE REVIEW

File: src/components/RadioButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+export const RadioButtons = ({handleChange}) => {
+  return (
+    <div className="radio-buttons-container">
+      <form className="label">
+        <span className="number">3.</span>
+        <span className="question">Do you prefer attending live concerts or listening to music at home?</span> 
+      </form>
+
+      <div className="mapContainer">
```
Comment (vittoriamatteoli) [inline src/components/RadioButtons.jsx:29]: ClassName could be more specific.

File: src/components/RangeSlider.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      <input
+        className="slider"
+        type="range"
+        min={1}
+        max={5}
+        step={1}
+        value={value}
+        onChange={onChange}
+        list="tickmarks"
+      />
```
Comment (vittoriamatteoli) [inline src/components/RangeSlider.jsx:27]: Really nice that you went on with the stretch goal! Although would be nice to show the user in real-time the value he selected. You could display the value from your slider in a <p> or <span> tag underneath the slider.

File: src/components/SurveyForm.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+  const handleDiscoveryChange = (event) => {
+    setSelectedDiscovery(event.target.value);
+  };
+
+  const handleRangeChange = (event) => {
+    setSelectedRange(event.target.value);
+  };
+
+  return (
```
Comment (vittoriamatteoli) [inline src/components/SurveyForm.jsx:81]: I like that you exercised a lot on creating components and passing props. However, you could even control the display/don't display behaviour of the result and form by simply using a ternary operator, something like this:
` {!formSubmitted ?(<Form/> ) : ( <Result/> )} `
 // Render the form if it's not submitted
// Render the Result if the form has been submitted
By doing this you could execute the logic inside one component only and reduce the amount of functions and conditional statements.

File: src/components/DropDown2.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { useRef } from "react";
+import "./DropDown.css";
+
+export const DropDown2 = ({handleChange}) => {
+
+    const dropdownRef = useRef(null);
```
Comment (vittoriamatteoli) [inline src/components/DropDown2.jsx:6]: I didn't understand what's this for.

File: src/components/TextInput.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { useState } from "react";
+import "./TextInput.css";
+
+export const TextInput = ({value, handleChange}) => {
```
Comment (vittoriamatteoli) [inline src/components/TextInput.jsx:4]: You are passing value and handlechange as props but value is never used.?

**Review by vittoriamatteoli — COMMENTED** [review]
Cornelia and Jing, I'm really impressed with how well-structured and organized your React code is! It's evident that both of you collaborated effectively on this pair-project, and your code meets all the requirements for the assignment. I've left a few small comments for you to review, but overall, it looks great!

One thought I had while reviewing the code is that you might consider using fewer components in your React app. Since the section receiving the changes is consistently the same (a form), you could potentially consolidate some components. However, I also recognize that is great practicing with props and components. Keep up the excellent work!

File: src/components/DropDown.jsx
Comment (AntonellaMorittu) [inline src/components/DropDown.jsx:1]: could be good to improve the files structure to group the similar components with .jsx and .css inside a folder, like creating a`Dropdown` folder to store DropDown.css, DropDown.jsx and DropDown2.jsx 👍

File: src/components/RadioButtons.css
Comment (AntonellaMorittu) [inline src/components/RadioButtons.css:1]: same for the RadioButtons component files, create a folder to group them 👍

**Review by AntonellaMorittu — APPROVED** [review]
Hello Cornelia & Jing and congrats with your Survey project! The app satisfies the requirements and lovethe theme. I left a few comments in the code, overall well done with responsiveness and accessibility, a tip is to prevent the user to go to the next question if current is unanswered. Anyway, you did a great job! 👏

**Pull request closed without merging**